src/node/ssh.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * Provides utilities for handling SSH connections
+ */
+import * as net from "net";
+import * as cp from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+import * as sshTypes from "ssh2";
+import * as nodePty from "node-pty";
+import { FileEntry, SFTPStream } from "ssh2-streams";
+import { localRequire } from 'vs/server/src/node/util';
+import { sanitizeProcessEnvironment } from 'vs/base/common/processes';
+import { IProcessEnvironment } from 'vs/base/common/platform';
+
+const ssh = localRequire<typeof import("ssh2")>("ssh2/lib/client");
+
+/**
+ * Fills out all of the functionality of SSH using node equivalents.
+ */
+export function fillSSHSession(accept: () => sshTypes.Session) {
+	let pty: nodePty.IPty | undefined;
+	let activeProcess: cp.ChildProcess;
+	let ptyInfo: sshTypes.PseudoTtyInfo | undefined;
+	const env: { [key: string]: string } = {};
+
+	const session = accept();
+
+	// Run a command, stream back the data
+	const cmd = (command: string, channel: sshTypes.ServerChannel): void => {
+		if (ptyInfo) {
+			// Remove undefined env vars and remove VSCode / code-server env vars
+			const env = Object.keys(process.env).reduce((prev, k) => {
+				if (process.env[k] !== undefined) {
+					prev[k] = process.env[k] as string;
+				}
+				return prev;
+			}, {} as IProcessEnvironment);
+			sanitizeProcessEnvironment(env as IProcessEnvironment);
+
+			pty = nodePty.spawn(command, [], {
+				cols: ptyInfo.cols,
+				rows: ptyInfo.rows,
+				env,
+			});
+			pty.onData((d) => channel.write(d));
+			pty.on("exit", (exitCode) => {
+				channel.exit(exitCode);
+				channel.close();
+			});
+			channel.on("data", (d: any) => pty!.write(d));
+			return;
+		}
+
+		const proc = cp.spawn(command);
```

Comment:
I think if you try to `spawn` something like `ls -la` it'll fail because it's looking for a file actually named `ls -la`. Passing `shell: true` in the options to make `spawn` use a shell should work since that's what `exec` does by default.